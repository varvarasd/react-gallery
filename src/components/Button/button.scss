.buttons__wrapper {
    width: 100%;
    display: flex;
    justify-content: space-between;
    border: 0.1px solid #cfcfcf;
    border-top: none;
}

.button {
    &.previous,
    &.next {
        width: 100%;
        padding: 20px;
        &:hover {
            background-color: #f0f0f0;
        }
        &:focus {
            outline: none;
            background-color: #f0f0f0;
        }
    }
    &.next {
        &:focus {
            border-left: 0.1px solid #cfcfcf;
        }
    }
    &.previous {
        &:focus {
            border-right: 0.1px solid #cfcfcf;
        }
    }
}