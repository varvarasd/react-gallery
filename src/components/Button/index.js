import React from 'react';
import './button.scss';

const Button = ({role, addClass, labelName, action, children}) => {
    const buttonClasses = addClass ? `button ${addClass}` : 'button';
    return <button 
        role={role}
        label={labelName}
        onClick={action} 
        className={buttonClasses}>
            {children}
        </button>;
};

export default Button;