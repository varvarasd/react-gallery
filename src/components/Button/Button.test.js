import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Button from '.';

describe('Button', () => {
    it('renders', () => {
        const { container } = render(<Button />);
        expect(container).toMatchSnapshot();
    });

    it('renders with given params', () => {
        const someFunction = jest.fn();

        const { container } = render(
                <Button addClass='some-class'
                    labelName='some-label'
                    action={() => someFunction()} />
            );

        expect(container.firstChild).toHaveClass('some-class');
        expect(container.firstChild.getAttribute('label')).toBe('some-label');

        fireEvent.click(container.firstChild);
        expect(someFunction).toHaveBeenCalledTimes(1);
    });

    it('renders children', () => {
        const { getByText } = render(<Button><div>button child</div></Button>);
        expect(getByText(/button child/)).toBeInTheDocument();
    });
})