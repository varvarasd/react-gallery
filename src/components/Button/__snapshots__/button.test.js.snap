// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders 1`] = `
<div>
  <button
    class="button"
  />
</div>
`;
